### PR TITLE
impr/logcli: Added label output filters + tests

### PR DESCRIPTION
**Pull request by CyrilPeponnet**
* Fix a bug in the `substract` / `intersect` and wrote some tests for it
* Added a filter for labels as follow:

```console
      --exclude-label=EXCLUDE-LABEL ...  
                         Exclude labels given the provided key during output.
      --include-label=INCLUDE-LABEL ...  
                         Include labels given the provided key during output.
      --labels-length=0  Set a fixed padding to labels
```

Here is the new output tested with query and tail:

```console

# Query
➜ logcli query '{app="influxdb"}'  --limit 2
2019/05/08 15:32:44 http://localhost:3100/api/prom/query?query=%7Bapp%3D%22influxdb%22%7D&limit=2&start=1557351164039536000&end=1557354764039536000&direction=BACKWARD&regexp=
Common labels: {app="influxdb", container_name="influxdb", job="default/influxdb", namespace="default", stream="stderr", type="database"}
2019-05-08T22:32:42Z {__filename__="/var/log/pods/dea836a3-6247-11e9-aa44-42010a800150/influxdb/0.log", pod="influxdb-2"} here are the logs`
2019-05-08T22:32:40Z {__filename__="/var/log/pods/cf3f655c-6247-11e9-aa44-42010a800150/influxdb/0.log", pod="influxdb-3"} here are the logs

# No labels
➜ logcli query '{app="influxdb"}'  --limit 2 --no-labels
2019/05/08 15:32:55 http://localhost:3100/api/prom/query?query=%7Bapp%3D%22influxdb%22%7D&limit=2&start=1557351175878144000&end=1557354775878144000&direction=BACKWARD&regexp=
Common labels: {app="influxdb", container_name="influxdb", job="default/influxdb", namespace="default", stream="stderr", type="database"}
2019-05-08T22:32:52Z  here are the logs
2019-05-08T22:32:50Z  here are the logs


# Strip __filename__
➜ logcli query '{app="influxdb"}'  --limit 2 --exclude-label=__filename__
2019/05/08 15:33:14 http://localhost:3100/api/prom/query?query=%7Bapp%3D%22influxdb%22%7D&limit=2&start=1557351194826225000&end=1557354794826225000&direction=BACKWARD&regexp=
Common labels: {app="influxdb", container_name="influxdb", job="default/influxdb", namespace="default", stream="stderr", type="database"}
Ignoring labels key: __filename__
2019-05-08T22:33:12Z {pod="influxdb-2"} here are the logs
2019-05-08T22:33:10Z {pod="influxdb-3"} here are the logs

## Ignore __filename__ and add app key label
➜ logcli query '{app="influxdb"}'  --limit 2 --exclude-label=__filename__ --include-label=app
2019/05/08 15:33:24 http://localhost:3100/api/prom/query?query=%7Bapp%3D%22influxdb%22%7D&limit=2&start=1557351204140870000&end=1557354804140870000&direction=BACKWARD&regexp=
Common labels: {container_name="influxdb", job="default/influxdb", namespace="default", stream="stderr", type="database"}
Ignoring labels key: __filename__
2019-05-08T22:33:22Z {app="influxdb", pod="influxdb-2"} here are the logs
2019-05-08T22:33:20Z {app="influxdb", pod="influxdb-3"} here are the logs


# Tail
➜ logcli query '{app="influxdb"}' -t
ws://localhost:3100/api/prom/tail?query=%7Bapp%3D%22influxdb%22%7D&regexp=
2019-05-08T22:33:35Z {__filename__="/var/log/pods/efae87a6-6247-11e9-aa44-42010a800150/influxdb/0.log", app="influxdb", container_name="influxdb", job="default/influxdb", namespace="default", pod="influxdb-1", stream="stderr", type="database"} here are the logs
2019-05-08T22:33:37Z {__filename__="/var/log/pods/0362f925-6248-11e9-aa44-42010a800150/influxdb/0.log", app="influxdb", container_name="influxdb", job="default/influxdb", namespace="default", pod="influxdb-0", stream="stderr", type="database"} here are the logs

# no labels
➜ logcli query '{app="influxdb"}'  --no-labels
2019/05/08 15:34:05 http://localhost:3100/api/prom/query?query=%7Bapp%3D%22influxdb%22%7D&limit=30&start=1557351245967880000&end=1557354845967880000&direction=BACKWARD&regexp=
Common labels: {app="influxdb", container_name="influxdb", job="default/influxdb", namespace="default", stream="stderr", type="database"}
2019-05-08T22:34:02Z  here are the logs
2019-05-08T22:34:00Z  here are the logs

# Strip __filename__
➜ logcli query '{app="influxdb"}'  -t --exclude-label=__filename__
ws://localhost:3100/api/prom/tail?query=%7Bapp%3D%22influxdb%22%7D&regexp=
Ingoring labels key: __filename__
2019-05-08T22:52:17Z {app="influxdb", container_name="influxdb", job="default/influxdb", namespace="default", pod="influxdb-0", stream="stderr", type="database"} here are the logs
2019-05-08T22:52:17Z {app="influxdb", container_name="influxdb", job="default/influxdb", namespace="default", pod="influxdb-3", stream="stderr", type="database"} here are the logs



# Show only app and pod labels
# Note the deffierent behaviour than query because we don't have common set of labels to use
➜ logcli query '{app="influxdb"}'  -t --label=app --include-label=pod
ws://localhost:3100/api/prom/tail?query=%7Bapp%3D%22influxdb%22%7D&regexp=
Print only labels key: app,pod
2019-05-08T22:51:33Z {app="influxdb", pod="influxdb-1"} here are the logs
2019-05-08T22:51:33Z {app="influxdb", pod="influxdb-3"} here are the logs
```

> Please note the different in behavior with `--include-label` between query and tail as tail doesn't have a common set of labels to strip, `--include-label` only print thoses labels instead of appending with query.

### CODE REVIEW

**Comment by daixiang0** [comment]
rename `no-label` as `skip-label` is better.

File: cmd/logcli/query.go [hunk 1]
```diff
@@ -21,9 +22,8 @@ func doQuery() {
 	}
 
 	var (
-		i            iter.EntryIterator
-		common       labels.Labels
-		maxLabelsLen = 100
```
Comment (daixiang0) [inline cmd/logcli/query.go]: Just remove it, how about put it into config?

File: cmd/logcli/query.go [hunk 2]
```diff
@@ -43,24 +43,43 @@ func doQuery() {
...
+	if len(*showLabelsKey) > 0 {
+		common = common.MatchLabels(false, *showLabelsKey...)
+	}
 
 	if len(common) > 0 {
 		fmt.Println("Common labels:", color.RedString(common.String()))
 	}
 
+	if len(*ignoreLabelsKey) > 0 {
+		fmt.Println("Ignoring labels key:", color.RedString(strings.Join(*ignoreLabelsKey, ",")))
```
Comment (daixiang0) [inline cmd/logcli/query.go]: call `log` and print error/debug info to stderr.

File: cmd/logcli/query.go (hunk 1 above)
Comment (CyrilPeponnet) [reply cmd/logcli/query.go]: There is no config file for logcli, I'm not following you.

**Comment by CyrilPeponnet** [comment]
@daixiang0 ready for another round.

**Pull request merged**